### ISSUE TITLE
Added/removed mass markers are not stored in Scenario.MasterChain

### ISSUE THREAD

**Issue by Uveso** [ozonexo3/FAForeverMapEditor#115]
Problems with AI and mass marker.
The script does not change the data inside Scenario.MasterChain._MASTERCHAIN_.Markers and the AI can't find the new mexes or is trying to build where a mex was removed.
The command CreateResourceDeposit and CreatePropHPR are registering the changes for the c-engine but i can't access the date from LUA.

The normal AIs (GPG/SOrian) are searching for a buildplace with
`location = aiBrain:FindPlaceToBuild(mass-spot-position)`
but the location is not the wanted mass spot, it's the next available spot registered inside the c-engine with CreateResourceDeposit/CreatePropHPR.

Without the extra FindPlaceToBuild() command it's not possible on adaptive maps to build on mass spots directly.
![AdaptiveMassmarkers](https://user-images.githubusercontent.com/17804547/71526042-15605000-28d5-11ea-80bc-e3561d145e5a.png)

**Comment by Uveso** [comment]
This is how an adaptive map script can add Mass/Hydro markers to the Scenario.MasterChain

1st change:
https://github.com/ozonexo3/FAForeverMapEditor/blob/master/Structure/AdaptiveMap/mapname_script.lua#L462
Add this to the if (doit) statement:
```
        elseif tblData.resource then
            --LOG('* AI-Debug: Deleting resource marker: '..name..' - '..repr(Scenario.MasterChain._MASTERCHAIN_.Markers[name]))
            Scenario.MasterChain._MASTERCHAIN_.Markers[name] = nil
        end
```
Overview:
```
       if (doit) then
            if(tblData.type == 'Mass') then
                spawnedMexNumber = spawnedMexNumber + 1
                spawnedMassSpots[spawnedMexNumber] = name
            end
            
            --enable 'forwardCrazyrushMexes'
            if crazyrush_mexes == 2 and FalseIfNotInList(name, forwardCrazyrushMexes, MassString, true) then
                spawnresource(tblData.position,tblData.type, false)
            else
                spawnresource(tblData.position,tblData.type, true)
            end
        elseif tblData.resource then
            --LOG('* AI-Debug: Deleting resource marker: '..name..' - '..repr(Scenario.MasterChain._MASTERCHAIN_.Markers[name]))
            Scenario.MasterChain._MASTERCHAIN_.Markers[name] = nil
        end
```
2nd change: https://github.com/ozonexo3/FAForeverMapEditor/blob/master/Structure/AdaptiveMap/mapname_script.lua#L738
Only add this line into the for loop:
```
            --LOG('* AI-Debug: creating resource Marker: '..i..' - '..repr(tblData))
            Scenario.MasterChain._MASTERCHAIN_.Markers[i] = tblData
```
The full loop:
```
    for i, tblData in pairs(markers) do
        if tblData.resource then
            spawnresource(tblData.position, tblData.type, true)
            --LOG('* AI-Debug: creating resource Marker: '..i..' - '..repr(tblData))
            Scenario.MasterChain._MASTERCHAIN_.Markers[i] = tblData
        end
    end
```
I also debugged this visual, so i am pretty sure it works like intended :)
![BugfixVisualDebug](https://user-images.githubusercontent.com/17804547/71588310-4d5bd300-2b21-11ea-986e-b2d45eea2d9e.png)

**Comment by Soft-2019** [comment]
hello, how are you able to use the editor version 605? i crash upon opening my maps all other versions work fine.

**Comment by Uveso** [comment]
Hello Soft-2019,

this bug `issue` ist reserved for the `dded/removed mass markers are not stored in Scenario.MasterChain` problem.

Please open a new issue in case you found a bug or contact ozonex on discord or make a post inside the forum:
http://forums.faforever.com/viewtopic.php?f=45&t=10647

**Comment by ozonexo3** [comment]
Implemented in v0.606
